### ISSUE TITLE
[iOS PostProcessBuild] Auto enable Background Mode -> Remote notifications 

### ISSUE THREAD

**Issue by GabrielCoriiu** [OneSignal/OneSignal-Unity-SDK#175]
Hi,

This is a suggestion, not an issue.
Please add this code as a post processing script to auto enable Background Mode -> Remote notifications whenever a build is made

```C
[PostProcessBuild]
public static void OnPostprocessBuild(BuildTarget buildTarget, string path)
{
	if (buildTarget == BuildTarget.iOS)
	{
		string projPath = PBXProject.GetPBXProjectPath(path);

		PBXProject proj = new PBXProject();
		proj.ReadFromString(File.ReadAllText(projPath));
		string targetName = PBXProject.GetUnityTargetName();
		string target = proj.TargetGuidByName(targetName);
		
		var projCapability = new ProjectCapabilityManager(projPath, "Unity-iPhone/mmk.entitlements", targetName);

		proj.AddCapability(target, PBXCapabilityType.BackgroundModes);
		projCapability.AddBackgroundModes(BackgroundModesOptions.RemoteNotifications);

		projCapability.WriteToFile();
		File.WriteAllText(projPath, proj.WriteToString());
	}
}
```

Cheers

**Comment by rgomezp** [comment]
Hello. Thanks for your suggestion. I will go ahead and close the issue though please note we will continue to track this internally as we consider it as a feature. Thanks again

**Comment by jkasten2** [comment]
@GabrielCoriiu Thanks for the code snipit. We implemented it in PR #251 and will be included in the next release

**Comment by AlexZonov** [comment]
This is bad code!

Resolve:
![image](https://user-images.githubusercontent.com/6861200/87324162-64e82f80-c538-11ea-871b-f2be2292a438.png)

**Comment by AlexZonov** [comment]
This is bad code!

Resolve:
```c#
[PostProcessBuild]
public static void OnPostprocessBuild(BuildTarget buildTarget, string path)
{
	if (buildTarget == BuildTarget.iOS)
	{
		string projPath = PBXProject.GetPBXProjectPath(path);

		PBXProject proj = new PBXProject();
		proj.ReadFromString(File.ReadAllText(projPath));
		string targetName = PBXProject.GetUnityTargetName();
		string target = proj.TargetGuidByName(targetName);
		
		proj.AddCapability(target, PBXCapabilityType.BackgroundModes);
                File.WriteAllText(projPath, proj.WriteToString());
		
		var projCapability = new ProjectCapabilityManager(projPath, "Unity-iPhone/mmk.entitlements", targetName);
                projCapability.AddBackgroundModes(BackgroundModesOptions.RemoteNotifications);
		projCapability.WriteToFile();
	}
}
```

**Comment by jonahgoldsaito** [comment]
Quick change to:

```
string guid = proj.GetUnityMainTargetGuid();
proj.AddCapability(guid, PBXCapabilityType.BackgroundModes);
```

**Comment by jkasten2** [comment]
This has been addressed in the 3.0.1 release. Please check out or migration guide to get updated to this version.
https://github.com/OneSignal/OneSignal-Unity-SDK/blob/main/MIGRATION.md
